src/wallet.ts

```diff
@@ -0,0 +1,218 @@
+import ethTx = require('ethereumjs-tx');
+import ethWallet = require('ethereumjs-wallet');
+
+declare const Buffer;
+declare const setTimeout;
+
+export default class Wallet {
+  length: number;
+  nonce: number;
+  web3: any;
+
+  constructor(web3: any) {
+    this.length = 0;
+    this.nonce = 0;
+    this.web3 = web3;
+  }
+
+  _findSafeIndex(pointer = 0) {
+    pointer = pointer;
+    if (this.hasOwnProperty(pointer)) {
+      return this._findSafeIndex(pointer + 1);
+    } else {
+      return pointer;
+    }
+  }
+
+  _currentIndexes() {
+    const keys = Object.keys(this);
+    const indexes = keys
+      .map((key) => parseInt(key, 10))
+      .filter((n) => n < 9e20)
+      .slice(0, this.length);
+    return indexes;
+  }
+
+  create(numAccounts) {
+    for (let i = 0; i < numAccounts; i++) {
+      const wallet = ethWallet.generate();
+      this.add(wallet);
+    }
+    return this;
+  }
+
+  add(wallet) {
+    if (!this[wallet.getAddressString()]) {
+      const idx = this._findSafeIndex();
+      wallet.index = idx;
+
+      this[idx] = wallet;
+      this[wallet.getAddressString()] = wallet;
+      this[wallet.getAddressString().toLowerCase()] = wallet;
+      this.length++;
+      return wallet;
+    } else {
+      return this[wallet.getAddressString()];
+    }
+  }
+
+  rm(addressOrIndex) {
+    const wallet = this[addressOrIndex];
+
+    if (wallet && wallet.getAddressString()) {
+      delete this[wallet.getAddressString()];
+      delete this[wallet.getAddressString().toLowerCase()];
+      delete this[wallet.index];
+      this.length--;
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  clear() {
+    const _this = this;
+    const indexes = this._currentIndexes();
+
+    indexes.forEach((idx) => {
+      _this.rm(idx);
+    });
+
+    return this;
```

Comment:
Same here, not sure why we return the whole object `this`.

Reply:
same answer

Reply:
These parts of the code came in from the web3 1.0.0 wallet implementation when we had to switch back to web3 0.20.* and lost some functionality. I basically just copied the code we needed over to here